app/src/main/java/com/example/mymovie/MainActivity.kt

```diff
@@ -3,20 +3,43 @@ package com.example.mymovie
 import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.View
+import android.widget.ImageView
 import android.widget.TextView
+import androidx.core.content.ContextCompat.startActivity
 
-class MainActivity : AppCompatActivity() {
+class MainActivity :
+        AppCompatActivity(R.layout.activity_main),
+        FragmentMoviesListClickListener,
+        BackToListClickListener {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        if (savedInstanceState == null) {
+            addMovieList()
+        }
+    }
 
-        val textView: TextView = findViewById(R.id.first_text_view)
-        textView.setOnClickListener { moveToNextScreen() }
+    private fun addMovieList() {
```

Comment:
addMovieList and toCardViewClicked are very similar
create some generic fun with parameter